### ISSUE TITLE
"Framework!ProfileDestroyer::~ProfileDestroyer()" crash when closing Tor windows

### ISSUE THREAD

**Issue by kjozwiak** [brave/brave-browser#8067]
## Description 

Originally found/reported by @yrliou. When closing opened Tor windows on `CR 80.0.3987.78`, the browser will crash via the following:

```
 Brave Browser Framework!ProfileDestroyer::~ProfileDestroyer() [profile_destroyer.cc : 168 + 0x0]
```

Seems like this is happening on `CR 80.0.3987.66` and `CR 80.0.3987.78` but **NOT** `CR 80.0.3987.85`

Crashes:

* https://stats.brave.com/dashboard#crash/5e3a28880dd633006dec6f43

@brave/legacy_qa lets wait till we get a build with `80.0.3987.85` and ensure Tor is working as expected.

## Steps to Reproduce

   1. install Brave that's running `CR 80.0.3987.66` or `CR 80.0.3987.78`
   2. open a Tor window
   3. close the Tor window

## Actual result:

When closing Tor windows, Brave will crash as described above.

## Expected result:

Brave shouldn't be crashing when closing Tor windows.

## Reproduces how often: 

100% reproducible using the above STR.

## Brave version (brave://version info)

Brave | 1.3.110 Chromium: 80.0.3987.78 (Official Build) (64-bit)
-- | --
Revision | cb5aa81cb0a3a2c674b93bc7ffb34a86cd3b4802-refs/branch-heads/3987@{#752}
OS | macOS Version 10.15.3 (Build 19D76)

## Version/Channel Information:

- Can you reproduce this issue with the current release? `N/A` (doesn't have CR80)
- Can you reproduce this issue with the beta channel? `Yes`
- Can you reproduce this issue with the dev channel? `Yes`
- Can you reproduce this issue with the nightly channel? `Yes`

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? `N/A`
- Does the issue resolve itself when disabling Brave Rewards? `N/A`
- Is the issue reproducible on the latest version of Chrome? N/A`

## Miscellaneous Information:

CCing @rebron @yrliou @brave/legacy_qa @mkarolin

**Comment by kjozwiak** [comment]
As per the above, @yrliou mentioned that this seems to be working under `CR 80.0.3987.85`. Created this issue so QA makes sure that Tor is working as expected once we get builds running CR `80.0.3987.85`.

**Comment by yrliou** [comment]
Tried with v1.3.111 (CR 80.0.3987.85) and didn't see any crashes when closing Tor window.

**Comment by kjozwiak** [comment]
Closing as per above 👍 Looks like it was addressed with `80.0.3987.85` as per https://github.com/brave/brave-browser/issues/8067#issuecomment-582215651. QA will go through the Tor manual passes and ensure everything else is working as expected.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `macOS 10.15.3 x64` using the following build:

Brave | 1.3.111 Chromium: 80.0.3987.85 (Official Build) (64-bit)
-- | --
Revision | 583b05dfa4ffc657f0f7c5cc13f53aa17c9a5bcf-refs/branch-heads/3987@{#791}
OS | macOS Version 10.15.3 (Build 19D76)

* Verified the original STR outlined under https://github.com/brave/brave-browser/issues/8067#issue-560106000
* Went through the cases outlined via https://github.com/brave/qa-resources/blob/master/wikitemplate.md#tor-tabs under https://github.com/brave/brave-browser/issues/8050
* ensured that opening/closing a Tor windows several times didn't cause any issues
* ensured that opening/closing a Guest windows several times didn't cause any issues
* ensured that opening/closing PB windows several times didn't cause any issues
